### PR TITLE
tests: Attempt to reproduce #15220 on timed progress notification

### PR DESCRIPTION
**Pull request by serathius**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.
cc @ahrtr @wojtek-t @scpmw

### CODE REVIEW

**Comment by serathius** [comment]
ref https://github.com/etcd-io/etcd/issues/15220

**Comment by serathius** [comment]
Also verified that there is no repro on v3.4.23

File: tests/linearizability/watch.go [hunk 1]
```diff
@@ -58,16 +58,27 @@ func collectClusterWatchEvents(ctx context.Context, t *testing.T, lg *zap.Logger
...
+			if !gotProcessNotify {
+				panic("Expected at least one notify")
+			}
 			return events
 		default:
 		}
-		for resp := range c.Watch(ctx, "", clientv3.WithPrefix(), clientv3.WithRev(lastRevision)) {
-			lastRevision = resp.Header.Revision
+		//for resp := range c.Watch(ctx, "", clientv3.WithPrefix(), clientv3.WithRev(lastRevision)) {
+		for resp := range c.Watch(ctx, "", clientv3.WithPrefix(), clientv3.WithRev(lastRevision), clientv3.WithProgressNotify()) {
```
Comment (ahrtr) [inline tests/linearizability/watch.go:75]: minor comment: do not add `clientv3.WithProgressNotify()` if `WatchProcessNotifyInterval == 0`

File: tests/linearizability/watch.go [hunk 2]
```diff
@@ -58,16 +58,27 @@ func collectClusterWatchEvents(ctx context.Context, t *testing.T, lg *zap.Logger
...
+			if resp.Header.Revision < lastRevision {
+				panic("Revision should never decrease")
+			}
+			if resp.Header.Revision == lastRevision && (lastResponseProgressNotify && !resp.IsProgressNotify()) {
```
Comment (ahrtr) [inline tests/linearizability/watch.go:79]:
```suggestion
			if resp.Header.Revision == lastRevision && !resp.IsProgressNotify() {
```

Comment (ahrtr) [reply tests/linearizability/watch.go:79]: It should be always regarded as an error as long as current response isn't progress notification.

**Comment by serathius** [comment]
Note: this is a draft PR to validate reproducibility of #15220. Not ready for merge or review.

**Pull request closed without merging**